modules/arcore/arcore_interface.cpp

```diff
@@ -0,0 +1,649 @@
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "arcore_interface.h"
+
+#include "core/image.h"
+#include "platform/android/java_godot_wrapper.h"
+#include "platform/android/os_android.h"
+#include "platform/android/thread_jandroid.h"
+#include "servers/visual/visual_server_globals.h"
+#include "servers/visual_server.h"
+
+// TODO remove these temporary includes once we add our GL_TEXTURE_EXTERNAL_OES support to our drivers
+#include "platform_config.h"
+#ifndef GLES3_INCLUDE_H
+#include <GLES3/gl3.h>
+#else
+#include GLES3_INCLUDE_H
+#endif
+
+/**
+	@author Bastiaan Olij <mux213@gmail.com>, Robert Hofstra <robert.hofstra@knowlogy.nl>
+	ARCore interface between Android and Godot
+**/
+
+StringName ARCoreInterface::get_name() const {
+	return "ARCore";
+}
+
+int ARCoreInterface::get_capabilities() const {
+	return ARVRInterface::ARVR_MONO + ARVRInterface::ARVR_AR;
+}
+
+int ARCoreInterface::get_camera_feed_id() {
+	if (feed.is_valid()) {
+		return feed->get_id();
+	} else {
+		return 0;
+	}
+}
+
+void ARCoreInterface::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("_resume"), &ARCoreInterface::_resume);
+	ClassDB::bind_method(D_METHOD("_pause"), &ARCoreInterface::_pause);
+}
+
+bool ARCoreInterface::is_initialized() const {
+	// if we're in the process of initialising we treat this as initialised...
+	return (init_status != NOT_INITIALISED) && (init_status != INITIALISE_FAILED);
+}
+
+void ARCoreInterface::_resume() {
+	if (ar_session == NULL) {
+
+		OS_Android *os_android = (OS_Android *)OS::get_singleton();
+
+		print_line("Godot ARCore: Getting environment");
+
+		// get some android things
+		JNIEnv *env = ThreadAndroid::get_env();
+
+		godot_java = os_android->get_godot_java();
+		jobject context = godot_java->get_application_context();
+		if (context == NULL) {
+			print_line("Godot ARCore: Couldn't get context");
+			init_status = INITIALISE_FAILED; // don't try again.
+			return;
+		}
+
+		print_line("Godot ARCore: Create ArSession");
+
+		// === ATTENTION!  ATTENTION!  ATTENTION! ===
+		// This method can and will fail in user-facing situations.  Your
+		// application must handle these cases at least somewhat gracefully.  See
+		// HelloAR Java sample code for reasonable behavior.
```

Comment:
This can be removed since the ArCoreSetupActivity handles the user facing situations.

Reply:
Cool